test/e2e/unjoin_test.go

```diff
@@ -0,0 +1,105 @@
+		kubeConfigPath := fmt.Sprintf("%s/.kube/%s.config", homeDir, clusterName)
+		controlPlane := fmt.Sprintf("%s-control-plane", clusterName)
+		clusterContext := fmt.Sprintf("kind-%s", clusterName)
+		policyNamespace := testNamespace
+		policyName := deploymentNamePrefix + rand.String(RandomStrLength)
+		deploymentNamespace := testNamespace
+		deploymentName := policyName
+
+		deployment := helper.NewDeployment(deploymentNamespace, deploymentName)
+		policy := helper.NewPropagationPolicy(policyNamespace, policyName, []policyv1alpha1.ResourceSelector{
+			{
+				APIVersion: deployment.APIVersion,
+				Kind:       deployment.Kind,
+				Name:       deployment.Name,
+			},
+		}, policyv1alpha1.Placement{
+			ClusterAffinity: &policyv1alpha1.ClusterAffinity{
+				ClusterNames: []string{clusterName},
+			},
+		})
+
+		ginkgo.It("Test unjoining not ready cluster", func() {
+
+			ginkgo.By(fmt.Sprintf("Creating cluster: %s", clusterName), func() {
+				err := createCluster(clusterName, kubeConfigPath, controlPlane, clusterContext)
+				gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+			})
+
+			ginkgo.By(fmt.Sprintf("Joinning cluster: %s", clusterName), func() {
+				karmadaConfig := karmadactl.NewKarmadaConfig(clientcmd.NewDefaultPathOptions())
+				opts := karmadactl.CommandJoinOption{
+					GlobalCommandOptions: options.GlobalCommandOptions{
+						DryRun: false,
+					},
+					ClusterNamespace:  "karmada-cluster",
+					ClusterName:       clusterName,
+					ClusterContext:    clusterContext,
+					ClusterKubeConfig: kubeConfigPath,
+				}
+				err := karmadactl.RunJoin(os.Stdout, karmadaConfig, opts)
+				gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+			})
+
+			ginkgo.By(fmt.Sprintf("Creating Deployment workload in cluster: %s", clusterName), func() {
+				framework.CreatePropagationPolicy(karmadaClient, policy)
+				framework.CreateDeployment(kubeClient, deployment)
+			})
+
+			ginkgo.By(fmt.Sprintf("diasble cluster %s ", clusterName), func() {
+				err := disableCluster(controlPlaneClient, clusterName)
+				gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+			})
+
+			karmadaConfig := karmadactl.NewKarmadaConfig(clientcmd.NewDefaultPathOptions())
+			opts := karmadactl.CommandUnjoinOption{
+				GlobalCommandOptions: options.GlobalCommandOptions{
+					DryRun: false,
+				},
+				ClusterNamespace:  "karmada-cluster",
+				ClusterName:       clusterName,
+				ClusterContext:    clusterContext,
+				ClusterKubeConfig: kubeConfigPath,
+				Wait:              options.DefaultKarmadactlCommandDuration,
+			}
+			err := karmadactl.RunUnjoin(os.Stdout, karmadaConfig, opts)
+			gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+			fmt.Printf("Check the deployment in cluster %s is exist", clusterName)
+			deployment, err := kubeClient.AppsV1().Deployments(testNamespace).Get(context.TODO(), deploymentName, metav1.GetOptions{})
+			gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+			gomega.Expect(deployment).ShouldNot(gomega.BeNil())
```

Comment:
I think about it again, it's weird to check whether deployment exists on the control plane. Our purpose is to check whether we can abort deleting workload when the member cluster is unready, this does not affect resources on the control plane. I think we can remove this check.